### PR TITLE
Add current selected item to the generated markup to the new multilev…

### PR DESCRIPTION
**Pull request by sneridagh**
…el navigation

### CODE REVIEW

**Comment by agitator** [comment]
@sneridagh we decided to not mark the current item within the generated markup so we wouldn't need to query the catalog for every item and to be able to cache the generated markup (in the future or as necessary)

**Comment by sneridagh** [comment]
@agitator OK I get your point, only point out some thoughts: Are you sure you'll have the use case where you want to cache the nav? What if a user adds a new section, or subsection? The invalidation logic might add more overhead than not to cache it. Maybe that use case is very narrow, but the new nav is fast enough now thanks to this PLIP for not having to cache it :)

Maybe my use case is also narrow, but I also thought that we might want to port a feature that was already in the current viewlet generation markup, whatever you decide will be fine (as long as one can override it easily).

**Comment by agitator** [comment]
performance, performance, performance... ;-)

For invalidation of the cache there is a counter in the portal_catalog one could check.

**Comment by ale-rt** [comment]
Maybe this issue can be more conveniently solved client side with some lines of javascript?

**Comment by petschki** [comment]
I definitely would like to see a current item marker in the markup too. You cannot cache the navigation globally anyways because the `show_excluded_items` setting depends on the context path too (see https://github.com/plone/plone.app.layout/pull/192) ... 

could it be that the strategy of caching has changed since this PR proposal, because I see only a `@plone.memoize.view.memoize` decorator for the `navtree` property and this in fact is not contextless ... so if you agree with me, I'd fixup this PR to add a marker to the markup.

/cc @ale-rt @agitator @sneridagh

**Comment by erral** [comment]
How does Plone mark the actual "current" navigation item? Because I can't find it in the relevant viewlet but I see the "current" class added in the final HTML. Is there some JS involved? 

How can I do it if I don't render any of Plone's JS in our theme?

**Comment by agitator** [comment]
@erral the js is in https://github.com/plone/mockup/blob/master/mockup/patterns/navigationmarker/pattern.js

**Comment by erral** [comment]
Thanks!

File: plone/app/layout/navigation/navtree.py [hunk 1]
```diff
@@ -354,3 +364,153 @@ def insertElement(itemPaths, item, forceInsert=False):
...
+            for it in res:
+                if context_path is not None:
+                    # Determine if it's current object
+                    is_current = context_path == it.getPath()
+                pathkey = self.navtree_path
+                entry = {
+                    'id': it['id'],
+                    'uid': None,
+                    'url': it['url'],
+                    'title': it['title'],
```
Comment (vincentfretin) [inline plone/app/layout/navigation/navtree.py]: Please make sure the Home tab is translated! See https://github.com/plone/plone.app.layout/pull/223

**Comment by thet** [comment]
@sneridagh the PR is outdated, but if it's still on the table then I'd like to see an option to switch the rendering of a current selected item class on/off again.
The default should be ``False`` (we have the JS solution per default).

I know that currently it wouldn't hit performance to render this class. But doing so prevents the navigation portlet from being cached for a specific amount of time, e.g. 10mins for non-editors.

For reference, we still have a performance issue with the nav: https://github.com/plone/Products.CMFPlone/issues/3035

**Pull request closed without merging**